[FLINK-34409][ci] Enables (most) tests that were disabled for the AdaptiveScheduler due to missing features

XComp opened the pull request:
## What is the purpose of the change

Increasing test coverage.

## Brief change log

* Removes annotations

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage. PR is temporarily enabling the AdaptiveScheduler 

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

MartijnVisser commented on tools/azure-pipelines/build-apache-repo.yml:
```diff
@@ -69,7 +69,7 @@ stages:
           e2e_pool_definition:
             vmImage: 'ubuntu-20.04'
-          environment: PROFILE="-Dflink.hadoop.version=2.10.2"
+          environment: PROFILE="-Dflink.hadoop.version=2.10.2 -Penable-adaptive-scheduler"
```
Why do we want to activate the adaptive-scheduler profile here?

XComp replied:
That's a debug commit which I added to the PR to have a CI run with the AdaptiveScheduler enabled. I'm not planning to merge this commit.

The goal is to have a CI run that actually contains the changes of this PR. That way, we can verify that the tests are actually enabled now (even before merging the PR).

MartijnVisser replied:
Oh wow I completely missed the debug commit 🤦 - Thanks :)

MartijnVisser reviewed (commented):
Should we check if any on the other tests that are currently disabled for AdaptiveScheduler would now work, e.g. ?

```
    if [[ ${PROFILE} != *"enable-adaptive-scheduler"* ]]; then
        run_test "Wordcount on Docker test (custom fs plugin)" "$END_TO_END_DIR/test-scripts/test_docker_embedded_job.sh dummy-fs"

        run_test "Run Kubernetes test" "$END_TO_END_DIR/test-scripts/test_kubernetes_embedded_job.sh"
        run_test "Run kubernetes session test (default input)" "$END_TO_END_DIR/test-scripts/test_kubernetes_session.sh"
        run_test "Run kubernetes session test (custom fs plugin)" "$END_TO_END_DIR/test-scripts/test_kubernetes_session.sh dummy-fs"
        run_test "Run kubernetes application test" "$END_TO_END_DIR/test-scripts/test_kubernetes_application.sh"
        run_test "Run kubernetes application HA test" "$END_TO_END_DIR/test-scripts/test_kubernetes_application_ha.sh"
        run_test "Run Kubernetes IT test" "$END_TO_END_DIR/test-scripts/test_kubernetes_itcases.sh"

        run_test "Running Flink over NAT end-to-end test" "$END_TO_END_DIR/test-scripts/test_nat.sh" "skip_check_exceptions"

        if [[ `uname -i` != 'aarch64' ]]; then
            # Skip PyFlink e2e test, because MiniConda and Pyarrow which Pyflink depends doesn't support aarch64 currently.
            run_test "Run kubernetes pyflink application test" "$END_TO_END_DIR/test-scripts/test_kubernetes_pyflink_application.sh"

            # Hadoop YARN deosn't support aarch64 at this moment. See: https://issues.apache.org/jira/browse/HADOOP-16723
            # These tests are known to fail on JDK11. See FLINK-13719
            if [[ ${PROFILE} != *"jdk11"* ]]; then
                run_test "Running Kerberized YARN per-job on Docker test (default input)" "$END_TO_END_DIR/test-scripts/test_yarn_job_kerberos_docker.sh"
                run_test "Running Kerberized YARN per-job on Docker test (custom fs plugin)" "$END_TO_END_DIR/test-scripts/test_yarn_job_kerberos_docker.sh dummy-fs"
                run_test "Running Kerberized YARN application on Docker test (default input)" "$END_TO_END_DIR/test-scripts/test_yarn_application_kerberos_docker.sh"
                run_test "Running Kerberized YARN application on Docker test (custom fs plugin)" "$END_TO_END_DIR/test-scripts/test_yarn_application_kerberos_docker.sh dummy-fs"
            fi
        fi
    fi
```

XComp commented:
> Should we check if any on the other tests that are currently disabled for AdaptiveScheduler would now work, e.g. ?

Yikes, that was actually my intention. But looks like I forgot to do another pass over the e2e test scripts. :-D Thanks. I will enable those as well. We should add a comment with the reason if we notice that they still fail. :+1:

JingGe commented on flink-tests/src/test/java/org/apache/flink/test/checkpointing/UnalignedCheckpointITCase.java:
```diff
@@ -105,7 +103,6 @@
  * </ul>
  */
 @RunWith(Parameterized.class)
-@Category(FailsWithAdaptiveScheduler.class) // FLINK-21689
```
Should FLINK-21689 also be added into the issue links of FLINK-34409, since this annotation has been removed?

XComp replied:
sure

XComp commented:
I went a bit naively into this task. :-D Some of the tests are still failing with the `AdaptiveScheduler`. I analyzed the test failures and added a proper reasoning for the future to document why the tests are disabled (if it was a conceptual issue).

XComp commented:
@flinkbot run azure

XComp commented:
I force-pushed a rebase to most-recent `master` w/o any other changes. I suspect that the [test failures](https://dev.azure.com/apache-flink/98463496-1af2-4620-8eab-a2ecc1a2e6fe/_build/results?buildId=574080 we were seeing in CI are related to FLINK-34420

XComp commented:
I fixed an indentation issue and rebased the branch. @MartijnVisser @JingGe can one of you go ahead and approve this one? So that we can finalize the efforts and merge this PR. I'm going to create backports as well to enable the tests in the release branches as well. I guess, increasing the test coverage also in those branches doesn't hurt.

snuyanzin commented:
should it be rebased because of FLINK-34718
to be sure that ci is green?

XComp commented:
> this thing from PR description is not clear to me
> if we temporarily enable it, what is the long term plan here?

I was referring to commit https://github.com/apache/flink/pull/24285/commits/d0ea2d70bffaef5a03669240564e148ab6ea14d3 that enables the AdaptiveScheduler profile to verify the test execution but won't be merged into `master`

XComp commented:
I rebased the branch to most-recent `master`. But it won't make much of a difference: The failed tests in CI are unrelated to the changes of this PR.

XComp commented:
I created FLINK-34921 to cover the one test failure which should be unrelated to this PR's change. The second test failure is related to FLINK-34643

snuyanzin reviewed (approved):
Thanks for clarification

XComp commented:
[CI with AdaptiveScheduler](https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=58505&view=results) looks good. Only FLINK-34643 is failing.
